pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -706,14 +704,22 @@ def _run_transaction(base, display):
 
         try:
             base.do_transaction(display)
-            exit_reason = "DNF done"
+            error_items = False
+            for tsi in base.transaction:
+                if tsi.state == libdnf.transaction.TransactionItemState_ERROR:
+                    log.error("The transaction item %s ended in error state.", tsi)
+                    error_items = True
+            if error_items:
+                display.error("The transaction has ended with some items in error state.")
+            else:
+                display.quit("DNF quit")
         except BaseException as e:  # pylint: disable=broad-except
-            log.error("The transaction has ended abruptly: %s", str(e))
-            exit_reason = str(e) + traceback.format_exc()
+            msg = "The transaction process has ended abruptly: {}\n{}".format(
+                str(e), traceback.format_exc())
+            display.error(msg)
         finally:
             log.debug("The transaction has ended.")
             base.close()  # Always close this base.
-            display.quit(exit_reason or "DNF quit")
```

Comment:
I would prefer not to change the order of the `base.close()` and `display.quit()` for purely paranoid reasons.